xdsl/dialects/builtin.py

```diff
@@ -1167,9 +1167,9 @@ def verify(self) -> None:
 class TupleType(ParametrizedAttribute, BuiltinAttribute):
     name = "tuple"
 
-    types: ParameterDef[ArrayAttr[Attribute]]
+    types: ParameterDef[ArrayAttr[TypeAttribute]]
```

Comment:
this is technically unnecessary but seems harmless. I needed to change this file because `list` is invariant, so I thought I'd change the base type also.